x/tokenomics/keeper/settle_pending_claims.go

```diff
@@ -48,285 +62,50 @@ func (k Keeper) SettlePendingClaims(ctx cosmostypes.Context) (
 	for ; expiringClaimsIterator.Valid(); expiringClaimsIterator.Next() {
 		claim, iterErr := expiringClaimsIterator.Value()
 		if iterErr != nil {
-			return settledResults, expiredResults, iterErr
-		}
-		numExpiringClaims++
-
-		if err = settlementContext.ClaimCacheWarmUp(ctx, &claim); err != nil {
-			return settledResults, expiredResults, err
-		}
-
-		var (
-			// These values are used for cores business logic; minting, slashing, etc.
-			proofRequirement prooftypes.ProofRequirementReason
-			claimeduPOKT     cosmostypes.Coin
-
-			// These values are used for basic validation, telemetry and observability.
-			numClaimRelays,
-			numClaimComputeUnits,
-			numEstimatedComputeUnits uint64
-		)
-
-		sessionId := claim.GetSessionHeader().GetSessionId()
-
-		// DEV_NOTE: Not every Relay (Request, Response) pair in the session is inserted into the tree.
-		// See the godoc for GetNumRelays for more delays.
-		numClaimRelays, err = claim.GetNumRelays()
-		if err != nil {
-			return settledResults, expiredResults, err
-		}
-
-		// Ensure that the number of relays claimed is greater than 0.
-		// CreateClaim message handler validation already ensures that the number of relays
-		// is greater than 0, but this is a sanity check to ensure that no invalid claims
-		// are processed in the settlement phase.
-		if numClaimRelays == 0 {
-			logger.Error(fmt.Sprintf(
-				"SHOULD NEVER HAPPEN: claim for session ID %q has 0 relays, skipping settlement",
-				sessionId,
-			))
-
-			// TODO_CONSIDERATION: Treat this claim as expired, since it has no relays.
-			// This would result in the Supplier being slashed for submitting a claim with 0 relays.
-			// DEV_NOTE: This scenario should theoretically never occur because claims are only
-			// created when the number of compute units exceeds zero.
-			// CreateClaim message handler should have rejected the claim in the first place.
-			k.proofKeeper.RemoveClaim(ctx, sessionId, claim.SupplierOperatorAddress)
+			k.discardFaultyClaim(ctx, claim, iterErr, logger)
 			continue
 		}
 
-		// DEV_NOTE: We are assuming that (numClaimComputeUnits := numClaimRelays * service.ComputeUnitsPerRelay)
-		// because this code path is only reached if that has already been validated.
-		numClaimComputeUnits, err = claim.GetNumClaimedComputeUnits()
+		claimProcessingContext, err := k.settleClaim(ctx, settlementContext, claim, logger)
 		if err != nil {
-			return settledResults, expiredResults, err
-		}
-
-		// Ensure that the number of compute units claimed is greater than 0.
-		// CreateClaim message handler validation already ensures that the number of compute units
-		// is greater than 0, but this is a sanity check to ensure that no invalid claims
-		// are processed in the settlement phase.
-		if numClaimComputeUnits == 0 {
```

Comment:
Ditto

Reply:
Stopping here but it seems like we removed some "SHOULD NEVER HAPPEN" validations for some reason.

Reply:
I removed these checks because the current refactor makes them unnecessary.

They were originally added to catch the error returned by `ProcessTokenLogicModules`, which would otherwise propagate and halt the chain.

The current refactor addresses those globally.

This behaviour was introduced in PR #1375.
![image](https://github.com/user-attachments/assets/2d455785-3193-497a-a4b9-6781aa6b059f)
https://github.com/pokt-network/poktroll/blob/main/x/tokenomics/keeper/token_logic_modules.go#L86